### ISSUE TITLE
External tempo sync request

### ISSUE THREAD

**Issue by Ashankan** [mixxxdj/mixxx#13789]
### Feature Description

I have been tempo syncing Mixxx with my DAW by looping an empty file in Deck-3 while sending tap tempo midi note from the DAW to the "tap tempo" and "set beat marker" function of the deck so that it matches up with the DAW.
This works, but can take a couple of bars to fully sync the beatgrid, pluss sometimes Deck-3 doesn't follow my instructions perfectly because I press some button or something which I don't notice.

I would like to see a 5th deck which is a tempo clock only, and its beatgrid appears on the top of the waveform window.
It should have the ability to receive and send midi-clock, preferably also send and recieve Ableton link, tempo change, start/pause clock, stop/reset (CUE) clock, and sync (match tempo to other decks).

Basically: make a fifth deck which plays a midi-clock either internally from Mixxx or from an external source instead of an audio file.

**Comment by daschuer** [comment]
We have that already on the hood, but have not yet exposed it to the user:
https://github.com/mixxxdj/mixxx/blob/fe9608f80f1cdb031a07fb1acc930899c5e93434/src/engine/sync/internalclock.h#L17

At the time of introductions we had some flaws in the use case. Not sure if the state has improved since then. It is worth a try. Maybe it fits to yours?

**Comment by daschuer** [comment]
A first step would be to expose the CO
`[InternalClock],bpm` in a skin.

**Comment by Ashankan** [comment]
"The Internal Clock should not be given any new sources of bpm clock.  If someone wants to write a Midi Clock source, it should be a separate Syncable object that can become leader"

If the internal clock is used as master clock it would be impossible to run two decks in sync in another tempo than the midi clock.
I believe my suggestion is the best solution for the case. It would add the possibility to sync with external gear without affecting any functionality of Mixxx.
